src/ctap/crypto_wrapper.rs

```diff
@@ -184,28 +177,40 @@ impl TryFrom<cbor::Value> for PrivateKey {
 
 impl From<ecdsa::SecKey> for PrivateKey {
     fn from(ecdsa_key: ecdsa::SecKey) -> Self {
-        PrivateKey::EcdsaKey(ecdsa_key)
+        PrivateKey::Ecdsa(ecdsa_key)
     }
 }
 
-/// Encrypts the given private key and relying part ID hash into a credential ID.
+/// Encrypts the given private key and relying party ID hash into a credential ID.
 ///
 /// Other information, such as a user name, are not stored. Since encrypted credential IDs are
-/// stored server-side, this information is already available (unecrypted).
+/// stored server-side, this information is already available (unencrypted).
 ///
 /// Also, by limiting ourselves to private key and RP ID hash, we are compatible with U2F for
 /// ECDSA private keys.
+///
+/// This is v1, and we write the following data for ECDSA (algorithm -7):
+/// - 16 bytes: initialization vector for AES-256,
+/// -  4 bytes: algorithm in big endian encoding,
+/// -  1 byte : version number
```

Comment:
I was expecting the version number to not be encrypted. Also this documentation is confusing. The way I understand the code, the format is:
- IV (16B)
- Payload encrypted by the IV above (80B):
    - Algo (4B)
    - Version (1B)
    - Padding (11B)
    - Private key (32B)
    - RP id hash (32B)
- Hmac of everything above (32B)

In other words I was expecting something like:
- Version (1B)
- IV (16B)
- Payload encrypted by IV (80B): same as above but without version
- Hmac of everything (32B)

This is one more byte in total.

Reply:
Ah right, having the version outside helps if we want to replace AES-CBC or HMAC. The 2 problems I saw with that:

1. I wanted credential IDs to be as indistinguishable from the outside as possible, so it doesn't help with fingerprinting. As an example, if you are using a different version as everyone else, relying parties could link your accounts together. You are probably already identifyable as an OpenSK user.
2. Someone could take a credential ID and downgrade the cryptography, without us noticing. I don't have a concrete attack in mind, but I thought it could be best practise to not allow that.

Might make sense to look at related work for best practises.

Reply:
The version number is now part of the HMAC, but not part of the encryption. This should leave the information leak as the only problem. At the same time, this is could be used to our advantage. Technically, clients could, whenever they forward credential IDs, warn you to re-create them because of outdated crypto.

Reply:
Indeed that could also be used as our advantage.